internal/registry/clientsecretrequest/rest.go

```diff
@@ -6,27 +6,79 @@ package clientsecretrequest
 
 import (
 	"context"
+	"crypto/rand"
+	"encoding/hex"
 	"fmt"
+	"io"
+	"strings"
 
+	"golang.org/x/crypto/bcrypt"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apiextensions-apiserver/pkg/registry/customresource/tableconvertor"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	genericvalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/api/validation/path"
 	metainternalversion "k8s.io/apimachinery/pkg/apis/meta/internalversion"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
 	"k8s.io/apiserver/pkg/registry/rest"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
 	"k8s.io/utils/trace"
 
 	clientsecretapi "go.pinniped.dev/generated/latest/apis/supervisor/clientsecret"
+	configv1alpha1clientset "go.pinniped.dev/generated/latest/client/supervisor/clientset/versioned/typed/config/v1alpha1"
+	"go.pinniped.dev/internal/oidcclientsecretstorage"
 )
 
-func NewREST(resource schema.GroupResource) *REST {
+// cost is a good bcrypt cost for 2022, should take about 250 ms to validate
+// this value is expected to be increased over time to match CPU improvements
+// thus it must be kept private because validation of client secrets cannot rely
+// on a cost that changes without some form client secret storage migration
+// TODO write a unit test that fails in 2023 to ask this to be updated to latest recommendation.
+const cost = 12
+
+// nolint: gochecknoglobals
```

Comment:
This needs an integration test.

Reply:
grep `as=Table` in the kube codebase should reveal headers, etc to get table output for a kubectl call.

Reply:
This is essential to work or the user will never see the secret that is generated (as the secret is ONLY shown to the user on create, there is NO WAY to get it at a later time!).

An example call:

```bash
kubectl create -o yaml -f - <<EOF | yq e -
apiVersion: clientsecret.supervisor.pinniped.dev/v1alpha1
kind: OIDCClientSecretRequest
metadata:
  name: client.oauth.pinniped.dev-foo
  namespace: supervisor
spec:
  generateNewSecret: true
EOF
```

s going to need to be in the documentation so users know how to get the secret values they create else they will be 😢

Reply:
output should look like:

| clientName | secret | numberOfSecrets |

Reply:
`kubetl` does not support output=table on create calls.